apps/dashboard/src/app/(main)/(protected)/projects/[projectId]/project-settings/page-client.tsx

```diff
@@ -25,65 +52,141 @@ export default function PageClient() {
+    [jwksUrl]
+  );
+
+  // Memoize renderInfoLabel callback
+  const renderInfoLabel = useCallback((label: string, tooltip: string) => (
     <div className="flex items-center gap-2">
       <span>{label}</span>
       <SimpleTooltip type="info" tooltip={tooltip}>
         <span className="sr-only">{`More info about ${label}`}</span>
       </SimpleTooltip>
     </div>
-  );
+  ), []);
 
-  // Get current owner team
-  const currentOwnerTeam = teams.find(team => team.id === project.ownerTeamId) ?? throwErr(`Owner team of project ${project.id} not found in user's teams?`, { projectId: project.id, teams });
+  // Memoize current owner team lookup
+  const currentOwnerTeam = useMemo(
+    () => teams.find(team => team.id === project.ownerTeamId) ?? throwErr(`Owner team of project ${project.id} not found in user's teams?`, { projectId: project.id, teams }),
+    [teams, project.ownerTeamId, project.id]
+  );
 
   // Check if user has team_admin permission for the current team
   const hasAdminPermissionForCurrentTeam = user.usePermission(currentOwnerTeam, "team_admin");
 
-  // Check if user has team_admin permission for teams
-  // We'll check permissions in the backend, but for UI we can check if user is in the team
-  const selectedTeam = teams.find(team => team.id === selectedTeamId);
+  // Memoize selected team lookup
+  const selectedTeam = useMemo(
+    () => teams.find(team => team.id === selectedTeamId),
+    [teams, selectedTeamId]
+  );
+
   const currentTeamMembers = currentOwnerTeam.useUsers();
-  const teamSettingsPath = `/projects?team_settings=${encodeURIComponent(currentOwnerTeam.id)}`;
 
-  const handleTransfer = async () => {
+  // Memoize team settings path
+  const teamSettingsPath = useMemo(
+    () => `/projects?team_settings=${encodeURIComponent(currentOwnerTeam.id)}`,
+    [currentOwnerTeam.id]
+  );
+
+  // Fix race condition with ref and improve error handling
+  const handleTransfer = useCallback(async () => {
     if (!selectedTeamId || selectedTeamId === project.ownerTeamId) return;
+    if (transferInProgressRef.current) return; // Prevent concurrent transfers
 
+    transferInProgressRef.current = true;
     setIsTransferring(true);
     try {
       await user.transferProject(project.id, selectedTeamId);
 
+      toast({
+        title: 'Project transferred successfully',
+        variant: 'success'
+      });
+
       // Reload the page to reflect changes
       // we don't actually need this, but it's a nicer UX as it clearly indicates to the user that a "big" change was made
       window.location.reload();
     } catch (error) {
       console.error('Failed to transfer project:', error);
-      alert(`Failed to transfer project: ${error instanceof Error ? error.message : 'Unknown error'}`);
+      toast({
+        title: 'Failed to transfer project',
+        description: error instanceof Error ? error.message : 'Unknown error',
+        variant: 'destructive'
+      });
```

Comment:
never use toasts for something that the user needs to see, use a dialog/alert instead. generally toasts are kind of an anti-pattern